website/content/en/v0.27.0/troubleshooting.md

```diff
@@ -128,6 +128,22 @@ Your security groups are not blocking you from reaching your webhook.
 This is especially relevant if you have used `terraform-eks-module` version `>=18` since that version changed its security
 approach, and now it's much more restrictive.
 
+If this issue happens when installing karpenter for the first time and the provisionners are defined using the `extraObjects` key from the values file, ensure that :
+ * The helm install/upgrade command has the `--wait` flag (or `wait: true` when using helmfile)
```

Comment:
Why do we need the wait flag? I thought helm just orchestrates the hooks in the background and the `wait` construct is more of a client-side thing?

Reply:
The wait ensures the karpenter pods are up before starting to create the helm hooks.
If the pods are not running, the webhook has no endpoints, resulting in the error message from the description.

Reply:
Is that what `--wait` does? Have you tried this without the `--wait`? I was under the impression that it's just a client-side foreground wait vs a background run of the hooks

Reply:
I didn't try without the wait option since the wait switch is taken from the install doc [here](https://karpenter.sh/v0.27.1/getting-started/getting-started-with-karpenter/#3-create-a-cluster), should be consistent.

I'm not a helm guru and lack time to dig up the code to have the final word, but [that's the expected behavior according to the doc](https://helm.sh/docs/intro/using_helm/#helpful-options-for-installupgraderollback).

Reply:
Got it, yes you are right. Otherwise, it won't wait for resource readiness before it applies the extraObjects